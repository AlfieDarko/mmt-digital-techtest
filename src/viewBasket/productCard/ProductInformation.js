import React from 'react'
import PropTypes from 'prop-types'
import { Stack, Heading, Text } from '@chakra-ui/core'
import { priceFormatter } from '../../utils'

const ProductInformation = ({ title, id, price, description }) => {
  return (
    <>
      <Stack spacing={2} justifyContent="space-between" flexDirection="column" >
        <Stack flexDirection="row" justifyContent="inherit">
          <Heading as="h2" fontSize={["md", "md", "lg", "xl"]} data-testid={`title-${id}`}>
            {title}
          </Heading>
          {/* AA WCAG Colour */}
          <Text pl={24} fontWeight="bold" as="span" color="#a2310d" data-testid={`price-${id}`}>{priceFormatter(price)}</Text>
        </Stack>
        <Text fontSize="sm" width="350px" data-testid={`description-${id}`}>
          {description}
        </Text>
      </Stack>
    </>
  )
}

ProductInformation.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired
}

export default ProductInformation
