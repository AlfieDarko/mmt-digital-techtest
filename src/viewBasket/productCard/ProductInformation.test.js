import ProductInformation from './ProductInformation'
import React from "react";
import { render } from '@testing-library/react'

describe('ProdutInformation', () => {
  it('Should match snapshots', () => {
    const { asFragment } = render(<ProductInformation title="Beer!" id="beer1" price={100} description="Nice German beverage" />);
    expect(asFragment()).toMatchSnapshot();
  });
});