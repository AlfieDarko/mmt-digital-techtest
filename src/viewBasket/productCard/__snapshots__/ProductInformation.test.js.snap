// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProdutInformation Should match snapshots 1`] = `
<DocumentFragment>
  <div
    class="css-1b6n4o1"
  >
    <div
      class="css-13mgw31"
    >
      <h2
        class="css-1i5vpt6"
        data-testid="title-beer1"
      >
        Beer!
      </h2>
      <span
        class="css-1uqan7e"
        data-testid="price-beer1"
      >
        Â£1.00
      </span>
    </div>
    <p
      class="css-18oyjk9"
      data-testid="description-beer1"
    >
      Nice German beverage
    </p>
  </div>
</DocumentFragment>
`;
